storage/buckets/buckets_test.go

```diff
@@ -645,3 +645,107 @@ func TestDelete(t *testing.T) {
 		t.Fatalf("deleteBucket: %v", err)
 	}
 }
+
+func TestRPO(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+
+	// Clean up bucket before running the test
+	if err := testutil.DeleteBucketIfExists(ctx, client, bucketName); err != nil {
+		t.Fatalf("Error deleting bucket: %v", err)
+	}
+
+	// Must create a bucket in dual-region to set turbo replication
+	location := "NAM4"
+	if err := client.Bucket(bucketName).Create(ctx, tc.ProjectID, &storage.BucketAttrs{Location: location}); err != nil {
+		t.Fatalf("Create bucket: %v", err)
+	}
+
+	testutil.WaitForBucketToExist(ctx, t, bucket)
+
+	// Test enable turbo replication:
+	if err := setRPOAsyncTurbo(ioutil.Discard, bucketName); err != nil {
+		t.Fatalf("setRPOAsyncTurbo: %v", err)
+	}
+	// Verify that RPO was set correctly
+	attrs, err := bucket.Attrs(ctx)
+	if err != nil {
+		t.Fatalf("Bucket(%q).Attrs: %v", bucketName, err)
+	}
+	if attrs.RPO != storage.RPOAsyncTurbo {
+		t.Errorf("setRPOAsyncTurbo: got %s, want %s", attrs.RPO, storage.RPOAsyncTurbo)
+	}
+
+	// Test get turbo replication enabled:
+	buf := new(bytes.Buffer)
+	if err := getRPO(buf, bucketName); err != nil {
+		t.Errorf("getRPO: %v", err)
+	}
+	// Verify that the correct value was printed
+	got := buf.String()
+	want := "RPO is ASYNC_TURBO"
+	if !strings.Contains(got, want) {
+		t.Errorf("getRPO: got %v, want %v", got, want)
+	}
+
+	// Test disable turbo replication:
+	if err := setRPODefault(ioutil.Discard, bucketName); err != nil {
+		t.Errorf("setRPODefault: %v", err)
+	}
+	// Verify that RPO was set correctly
+	attrs, err = bucket.Attrs(ctx)
+	if err != nil {
+		t.Fatalf("Bucket(%q).Attrs: %v", bucketName, err)
+	}
+	if attrs.RPO != storage.RPODefault {
+		t.Errorf("setRPODefault: got %s, want %s", attrs.RPO, storage.RPODefault)
+	}
+}
+
+func TestCreateBucketTurboReplication(t *testing.T) {
```

Comment:
Why is this separate from the other samples' test?

Reply:
I merged it with the other RPO tests, all in one place now